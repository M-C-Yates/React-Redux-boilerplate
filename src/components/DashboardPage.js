import React from 'react';


const DashboardPage = () => (
  <React.Fragment>
    
  </React.Fragment>
);

export default DashboardPage;
