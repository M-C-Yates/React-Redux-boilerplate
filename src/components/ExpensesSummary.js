
import React from 'react';
import getExpensesTotal  from '../selectors/expenses-total';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = ({ expenseCount, expensesTotal}) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <p>Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></p>  
    </div>
  );
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: getExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);