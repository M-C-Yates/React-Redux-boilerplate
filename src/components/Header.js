import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="header__box">
      <Link className="header__title" to="/dashboard" exact={true}><h1 >Budget</h1></Link>
      <button className="btn btn--logout" onClick={startLogout}>Logout</button>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);


