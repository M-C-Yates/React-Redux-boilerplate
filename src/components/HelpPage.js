import React from 'react';

const HelpPage = () => (
  <React.Fragment>
    nothing here yet
  </React.Fragment>
);

export default HelpPage;
