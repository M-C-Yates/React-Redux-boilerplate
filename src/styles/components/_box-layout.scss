.box-layout {
  height: 95vh;
  width: 100%;
  margin-left: -1rem;
  display: flex;
  align-items: center;
  justify-content: center;


  &__box {
    background: fade-out($color: $color-primary, $amount: .1);
    color: #fdfdfd; 
    padding: $l-size $m-size;
    text-align: center;
    border-radius: 3px;
    width: 25rem;
    box-shadow: 0.5rem 1rem 5rem rgba(0, 0, 0, 0.219);

    display: flex;
    flex-direction: column;
  }

  &__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
  }
}