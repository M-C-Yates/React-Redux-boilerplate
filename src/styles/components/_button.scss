.btn {
    background-color: #B670CB;
    border: 1px solid transparent;
    color: $light-grey;
    margin: $s-size;
    margin-top: 2.5rem;
    padding: 0.5rem 2rem;
    border-radius: 3px;
    font-weight: 300;
    font-size: 2rem;
    // box-shadow: 1rem 1rem 3rem rgba(0, 0, 0, .1);

  &--login {
    margin: $s-size;
    padding: 1.5rem 1.5rem;
  }


  &--logout {
    background-color: transparent;
    font-size: 3rem;
  }

  &:hover,
  &:active {
    background-color: transparent;
    border: 1px solid $light-grey;
    color: $light-grey;
    transition: .5s all ease-out;
  }
}

.btn-link {
  background-color: #B670CB;
  border: 1px solid transparent;
  color: $light-grey;
  margin: $s-size;
  margin-top: 2.5rem;
  padding: 0.5rem 2rem;
  border-radius: 3px;
  font-weight: 300;
  font-size: 2rem;
  box-shadow: 0.5rem 0.5rem 1rem rgba(0, 0, 0, .1);
  
  &:hover,
  &:active {
    background-color: #a34bbd;
    transition: .5s all ease-out;
  }
}

.btn-danger {
  @extend .btn-link;
  margin-top: -1rem;
  background-color: #FF1744;

  &:hover,
  &:active {
    background-color: rgb(192, 17, 52);
    transition: .5s all ease-out;
  }
}