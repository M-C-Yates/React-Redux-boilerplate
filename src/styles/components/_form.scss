.form {
  display: flex;
  flex-direction: column;
  margin-top: $l-size;
  

  &__error {
    margin: 0%;
    margin-bottom: $m-size;
    font-style: italic;
  }

  > * {
    margin-bottom: $m-size;
  }
}