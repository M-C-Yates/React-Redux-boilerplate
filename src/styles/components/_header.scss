.header {
  background-color: $color-primary;
  height: 100%;
  border-top-left-radius: 2px;
  border-top-right-radius: 2px;
  padding-bottom: 1rem;

  
  &__box {
    display: flex;
    justify-content: space-evenly;
    align-items: baseline;
  }

  &__title {
    color:  $light-grey;
    text-decoration: none;
    
    h1 {
      margin: 0;
    }
  }
}