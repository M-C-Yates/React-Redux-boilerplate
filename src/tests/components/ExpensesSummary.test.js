import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should correctly render ExpensesSummary component', () => {
  const wrapper = shallow(<ExpensesSummary />);
  expect(wrapper).toMatchSnapshot();
});