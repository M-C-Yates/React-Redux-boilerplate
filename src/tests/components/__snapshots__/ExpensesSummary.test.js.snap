// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpensesSummary component 1`] = `
<div>
  <p>
    Viewing 
    <span />
     
    expenses
     totalling 
    <span>
      $0.00
    </span>
  </p>
</div>
`;
